src/table/src/requests.rs

```diff
@@ -49,7 +55,83 @@ pub struct CreateTableRequest {
     pub region_numbers: Vec<u32>,
     pub primary_key_indices: Vec<usize>,
     pub create_if_not_exists: bool,
-    pub table_options: HashMap<String, String>,
+    pub table_options: TableOptions,
+}
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq, Eq)]
+pub struct TableOptions {
+    /// Memtable size of memtable.
+    pub write_buffer_size: Option<ReadableSize>,
+    /// Time-to-live of table. Expired data will be automatically purged.
+    #[serde(with = "humantime_serde")]
+    pub ttl: Option<Duration>,
+    /// Extra options that may not applicable to all table engines.
+    pub extra_options: HashMap<String, String>,
+}
+
+pub const WRITE_BUFFER_SIZE_KEY: &str = "write_buffer_size";
+pub const TTL_KEY: &str = "ttl";
+
+impl TryFrom<&HashMap<String, String>> for TableOptions {
```

Comment:
Recommendation:
```
impl TryFrom<HashMap<String, String>> for TableOptions
```
I think it's more natural to consume the `value` after the try_form(value) is called